pkg/sql/catalog/catpb/privilege.go

```diff
@@ -513,7 +513,10 @@ func (p PrivilegeDescriptor) CheckPrivilege(user username.SQLUsername, priv priv
 		return user.IsNodeUser()
 	}
 
-	if privilege.ALL.IsSetIn(userPriv.Privileges) {
+	if privilege.ALL.IsSetIn(userPriv.Privileges) && priv != privilege.NOSQLLOGIN {
```

Comment:
Is there a reason why we have a "negative" privilege? It should probably be reversed to be positive to avoid requiring special casing like this since I think it makes the code hard to reason about.

Reply:
probably the reason is that unlike other privileges, we want people to be able to log in by default, and storing and managing a SQLLOGIN privilege for every user would have its own set of complexities that would make the code hard to reason about.

our team wasn't actually the one who added this privilege, so I don't know the specific reason.

but regardless, it would be a very expensive project for us to remove the old privilege and migrate to a new inverse one. and even apart from engineering effort, it would be a disruptive change for operators of CRDB. I don't think the amount of effort of that project is worth it compared to adding a special case to this privilege check.

Reply:
Ok I approved this since it looks like it addresses the issue we have currently, but I think we need a better long term fix in order to prevent this from causing future issues.

Reply:
in a followup PR i will make a way to make privileges be marked as negative, so it is easy to handle them all in the same way. but i don't want to backport that, since it will lead to more merge conflicts.